driver-mapping/src/test/java/com/datastax/driver/mapping/MapperSaveNullFields22Test.java

```diff
@@ -0,0 +1,154 @@
+/*
+ *      Copyright (C) 2012-2015 DataStax Inc.
+ *
+ *   Licensed under the Apache License, Version 2.0 (the "License");
+ *   you may not use this file except in compliance with the License.
+ *   You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *   Unless required by applicable law or agreed to in writing, software
+ *   distributed under the License is distributed on an "AS IS" BASIS,
+ *   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *   See the License for the specific language governing permissions and
+ *   limitations under the License.
+ */
+package com.datastax.driver.mapping;
+
+import com.datastax.driver.core.BoundStatement;
+import com.datastax.driver.core.CCMTestsSupport;
+import com.datastax.driver.core.utils.CassandraVersion;
+import com.datastax.driver.mapping.Mapper.Option;
+import com.datastax.driver.mapping.annotations.PartitionKey;
+import com.datastax.driver.mapping.annotations.Table;
+import org.testng.annotations.BeforeMethod;
+import org.testng.annotations.Test;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+@SuppressWarnings("unused")
+public class MapperSaveNullFields22Test extends CCMTestsSupport {
```

Comment:
`@CassandraVersion("2.2.0")` just sets the minimum version requirement of the test, so `MapperSaveNullFieldsTest` will fail when run against >= C* 2.2.  We have an internal ci test matrix that runs against all of the latest cassandra versions (1.2.19, 2.0.17, 2.1.17, 2.2.9, 3.0.13 and 3.10 at the moment).

I think what we could do instead is change `MapperSaveNullFieldsTest#should_save_null_fields` to change the criteria of the validation based on the protocol version in use, using something like this to get at the protocol version:

```java
        boolean unsetSupported = session().getCluster().getConfiguration().getProtocolOptions()
                .getProtocolVersion().compareTo(ProtocolVersion.V4) >= 0;
```

and then we could do something like this in `should_save_null_fields`:

```java
        if (nullName && !saveExpected) {
            if (unsetSupported) { // column should be present if unset supported and value should be unset
                assertThat(queryString).as(description).contains("name");
                assertThat(!bs.isSet("name"));
            } else {
                assertThat(queryString).as(description).doesNotContain("name");
            }
        } else {
            assertThat(queryString).as(description).contains("name");
        }

        if (nullPhone && !saveExpected) {
            if (unsetSupported) {
                assertThat(queryString).as(description).contains("phone");
                assertThat(!bs.isSet("phone"));
            } else {
                assertThat(queryString).as(description).doesNotContain("phone");
            }
        } else {
            assertThat(queryString).as(description).contains("phone");
        }
```

I concede that is a bit complex, but it'll make it so we can use the existing tests to validate both cases.